private/model/api/codegentest/service/awsquerycompatible/awsquerycompatible_test.go

```diff
@@ -0,0 +1,20 @@
+//go:build go1.8 && codegen
+// +build go1.8,codegen
+
+package awsquerycompatible
```

Comment:
This test is good to assert the behavior of the generated aliases, but it would also be helpful to have a unit test that asserts the mapping behavior of the API client is correct too.

The following is an example test that could be added to this test client. (Note the build tag is updated to go1.13 instead of go1.8. This is needed for the `errors.As` utility.

```go
//go:build go1.13
// +build go1.13

package awsquerycompatible

import (
	"errors"
	"io"
	"io/ioutil"
	"net/http"
	"strings"
	"testing"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/awserr"
	"github.com/aws/aws-sdk-go/aws/corehandlers"
	"github.com/aws/aws-sdk-go/aws/request"
	"github.com/aws/aws-sdk-go/awstesting/unit"
)

func TestAWSQuery(t *testing.T) {
	cases := map[string]struct {
		statusCode   int
		responseBody interface {
			io.Reader
			Len() int
		}
		headers         http.Header
		expectErrorCode string
	}{
		"legacy query code": {
			statusCode:      500,
			responseBody:    strings.NewReader(`{"__type":"com.amazonaws.sqs#QueueDeletedRecently", "message":"Some user-visible message"}`),
			expectErrorCode: "AWS.SimpleQueueService.QueueDeletedRecently",
		},
		"new error code": {
			statusCode:      500,
			responseBody:    strings.NewReader(`{"__type":"com.amazonaws.sqs#QueueNameExists", "message":"Some user-visible message"}`),
			expectErrorCode: "QueueNameExists",
		},
	}

	for name, c := range cases {
		t.Run(name, func(t *testing.T) {
			client := New(unit.Session.Copy(), &aws.Config{
				MaxRetries: aws.Int(0),
			})

			client.Handlers.Send.Swap(corehandlers.SendHandler.Name, request.NamedHandler{
				Name: corehandlers.SendHandler.Name,
				Fn: func(r *request.Request) {
					r.HTTPResponse = &http.Response{
						StatusCode: c.statusCode,
						ContentLength: func() int64 {
							if c.responseBody == nil {
								return 0
							}
							return int64(c.responseBody.Len())
						}(),
						Header: func() http.Header {
							if c.headers == nil {
								return http.Header{}
							}
							return c.headers
						}(),
						Body: ioutil.NopCloser(c.responseBody),
					}
				},
			})

			_, err := client.CreateQueueRequest(&CreateQueueRequestInput{
				QueueName: aws.String("queueName"),
			})
			if err == nil {
				t.Fatalf("expect error, got none")
			}

			var awsErr awserr.RequestFailure
			if !errors.As(err, &awsErr) {
				t.Fatalf("expect RequestFailure error, got %#v", err)
			}

			if e, a := c.expectErrorCode, awsErr.Code(); e != a {
				t.Errorf("expect %v code, got %v", e, a)
			}
		})
	}
}
```

Reply:
This unit test will require https://github.com/aws/aws-sdk-go/pull/4525 to be merged in since it fixes the SDK's handling of leading and trailing spaces of XML based error code and message values.